storm-core/src/jvm/backtype/storm/messaging/netty/Client.java

```diff
@@ -182,7 +177,7 @@ private boolean connectionEstablished(Channel channel) {
         // See:
         // - http://netty.io/3.9/api/org/jboss/netty/channel/ChannelEvent.html
         // - http://stackoverflow.com/questions/13356622/what-are-the-netty-channel-state-transitions
-        return channel != null && channel.isConnected();
+        return channel != null && channel.isOpen();
```

Comment:
Thanks for the code review. It could be a bug. Netty 4.x has simplified the state model. See 

http://netty.io/wiki/new-and-noteworthy-in-4.0.html#wiki-h4-19  

So channelOpen, channelBound, and channelConnected have been merged to channelActive and Channel.isBound() and isConnected() have been merged to isActive().  The isConnected() method is removed.  I think I should use isActive() instead of isOpen().  The code comments need update too.  I will fix it and other format issues this weekend if I get time.